src/index.js

```diff
@@ -0,0 +1,126 @@
+/*
+  MIT License http://www.opensource.org/licenses/mit-license.php
+  Author Tobias Koppers @sokra
+*/
+import ConstDependency from 'webpack/lib/dependencies/ConstDependency';
+import NullFactory from 'webpack/lib/NullFactory';
+import MissingLocalizationError from './MissingLocalizationError';
+
+/**
+ *
+ * @param {object|function} localization
+ * @param {object|string} Options object or obselete functionName string
+ * @constructor
+ */
+class I18nPlugin {
+  constructor(localization, options) {
+    // Backward-compatiblility
+    if (typeof options === 'string') {
+      options = {
+        functionName: options,
+      };
+    }
+
+    if (arguments[2]) {
+      options.failOnMissing = arguments[2];
+    }
+
+    this.options = options || {};
+    this.localization = localization ? (typeof localization === 'function' ? localization : makeLocalizeFunction(localization, !!this.options.nested)) : null;
+    this.functionName = this.options.functionName || '__';
```

Comment:
`functionName` => `name || placeholder`?
`failOnMissing` => `throw || fail || error` ?
`hideMessage`=> `silent || warn || message|| hide` ?

Since it's semver major anyways 😛